vioserial/sys/Control.c

```diff
@@ -43,23 +43,18 @@ VIOSerialSendCtrlMsg(
     sg.physAddr = MmGetPhysicalAddress(&cpkt);
     sg.length = sizeof(cpkt);
 
-    WdfSpinLockAcquire(pContext->CVqLock);
+    WdfWaitLockAcquire(pContext->COutVqLock, NULL);
     if(0 <= virtqueue_add_buf(vq, &sg, 1, 0, &cpkt, NULL, 0))
     {
         virtqueue_kick(vq);
         while(!virtqueue_get_buf(vq, &len))
         {
            KeStallExecutionProcessor(50);
```

Comment:
I wonder if we can use KeDelayExecutionThread now that the loop executes at passive IRQL.

Reply:
It's not a crime to call KeStallExecutionProcessor on PASSIVE_LEVEL for small delay. But I left it there not for delay but for calling pause instruction, because Windows compiler has no intrinsic for it. Now I think using KeDelayExecutionThread with 0 time interval would be a better solution. Thanks, Ladi!